### ISSUE TITLE
The new version template system erroneously injects newlines

### ISSUE THREAD

**Issue by djosborne** [projectcalico/calico#532]
The new version template system (https://github.com/projectcalico/calico/pull/515) adds newlines everywhere an `include` statement is used. That means everywhere we `{% include version component="calico_node" %}`, we see a newline after that content.

Example of an offender is here: http://docs.projectcalico.org/master/getting-started/docker/installation/manual

- In section `1. Download the calicoctl binary` you'll see an extra newline between the two code lines
- in `Check that calico/node is now running` you'll see a newline after `quay.io/calico/node:master`

Supposedly the liquid template engine added a `-` character (e.g. `{%- include version component="calico_node" -%}` to remove this whitespace, which may be a solution to this problem. But we'd need to investigate if github pages uses a version of jekyll which uses a version of liquid with that fix.

**Comment by tmjd** [comment]
I ran into this today too.  I don't believe the version that github uses has any way to suppress or strip the newline, either using `{%- ... -%}` or a `{% spaceless %}` tag I came across.  I don't think it is available in Liquid until version 4 and github pages is at version 3.0.3. https://pages.github.com/versions
Here is someone from github commenting that Liquid 4 needs to get into Jekyll core and then be available 1-2 months after https://github.com/github/pages-gem/issues/411 and that was 9 days ago.

This is definitely problematic as we can't use it in any place that the include value is used in the middle of a string, like in a Vagrantfile that ends up using the value in the middle of a command or part of a url.
